.github/workflows/main.yml

```
@@ -0,0 +1,61 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  test:
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+    
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-18.04]
+        python-version: [2.7]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Install test dependencies
+        run: |
+         pip install pyroma
+         pip install check-manifest
+         pip install twine
+         pip install wheel
+         pip list
+         
+      - name: Compile code
+        run: python -m compileall -f .
+          
+      - name: Run unit tests
+        run: test/test.py --unit --exit-early
+        
+      - name: Run integration tests
+        run: test/test.py
```

Is there a specific reason why you removed the `--exit-early` option here?

```suggestion
        run: test/test.py --exit-early
```

---

I am just considering it to be more informative to get multiple error messages at once, in case there are errors. And the tests are not that long running, to make the exits necessary.

But if you disagree, I can add the flag, no problem.

---

I have looked through the test code again, it should be fine for the integration tests to skip `--exit-early`. Only the unit tests require `--exit-early` to return a non-zero exit code on errors.

I agree with you that the time for running the tests is okay and getting more feedback is a good thing.